Dockerfile

```diff
@@ -1,5 +1,24 @@
-FROM ruby:2.5.1
-RUN apt-get update -qq && apt-get install -y ruby-dev
+FROM alpine:3.8
+
+LABEL org.label-schema.maintainer="Puppet Bolt Team <team-direct-change-bolt@puppet.com>" \
+      org.label-schema.vendor="Puppet" \
+      org.label-schema.url="https://github.com/puppetlabs/bolt" \
+      org.label-schema.name="PE Bolt Server" \
+      org.label-schema.license="Apache-2.0" \
+      # Is there a way to set this?
+      #org.label-schema.version=$PUPPET_VERSION \
+      org.label-schema.vcs-url="https://github.com/puppetlabs/bolt" \
+      # Same with these
+      #org.label-schema.vcs-ref="b75674e1fbf52f7821f7900ab22a19f1a10cafdb" \
+      #org.label-schema.build-date="2018-05-09T20:10:01Z" \
+      #org.label-schema.schema-version="1.0" \
+      org.label-schema.dockerfile="/Dockerfile"
+
+RUN \
+apk update && apk upgrade && \
+apk --no-cache add build-base ruby-dev git && \
```

Comment:
I think we should probably avoid having these in the final image. How about a multi-stage build that only copies over the installed gems? That should help with target size. I think we can have a final image < 100MB.

Reply:
Would that only copy the gems so that we don't need bundler, or also copy in ruby? I'm not opposed to it, but I'm not familiar with that pattern around containers.

Reply:
It can. See https://docs.docker.com/develop/develop-images/multistage-build/. The stages all go in a single Dockerfile, and are built the same way.

I'd suggest 3 stages: a base runtime image with ruby and openssl, a build image that installs all the gems (and builds their extensions), and the final image that copies over the installed gems (so we can avoid all the build tools and dependencies).